core/db/migrate/20150609093816_increase_scale_on_pre_tax_amounts.rb

```diff
@@ -0,0 +1,16 @@
+class IncreaseScaleOnPreTaxAmounts < ActiveRecord::Migration
+  def change
+    # set pre_tax_amount on shipments to discounted_amount - included_tax_total
+    # so that the null: false option on the shipment pre_tax_amount doesn't generate
+    # errors.
+    #
```

Comment:
I've been going back and forth these days on whether to use SQL or to use AR for migrations. For me, I would have done:

```ruby
Spree::Shipment.where(pre_tax_amount: nil).find_each do |shipment|
  pta = shipment.cost + shipment.promo_total - shipment.included_tax_total
  shipment.update_attribute 'pre_tax_amount', pta
end
```

This way its all inside of AR and none of the callbacks happen; I think. If the table name may change, then maybe:

class Shipment < Spree::Base
  self.table_name = 'spree_shipments'
end

Shipment.find_each do |shipment|
  ...
end